docs/flow/modin/backends/index.rst

```diff
@@ -0,0 +1,46 @@
+:orphan:
+
+Query compiler
+==============
+
+Modin has several execution backends. Calling any DataFrame API function will end up in
+some backend-specific method. Query compiler is a bridge between Modin DataFrame and
+an actual execution engine.
+
+.. image:: /img/simplified_query_flow.svg
+    :align: right
+    :width: 300px
+
+Query compilers of all backends provide a common API, which is used by Modin DataFrame
+to emulate pandas API. The role of the query compiler is to translate its API into
+a set of DataFrame algebra operations. Each query compiler instance contains
+:doc:`frame </flow/modin/engines/base/frame/index>` of the selected execution engine and query
+it with the compiled queries to get the result. The query compiler object is immutable,
+so the result of every method is a new query compiler.
+
+Query compilers API is defined by the :doc:`BaseQueryCompiler <base/query_compiler>` class
+and partially mimic the pandas API, however, they're not equal. Query compilers API
+is significantly reduced in comparison with pandas, since many corner cases or even the
+whole methods can be handled at the DataFrame level with the existing API.
+
+Query compiler is the level where Modin stops distinguishing Frame and Series objects.
```

Comment:
```suggestion
The query compiler is the level where Modin stops distinguishing Frame and Series (or column) objects.
```

Reply:
just a bit of curiosity, what does "Series **or column**" means?

Reply:
I believe @devin-petersohn mean that we treat Series (or column - column of a DataFrame is a Series object as well) as DataFrame at the QC layer.